numpy/random/src/distributions/distributions.c

```diff
@@ -121,15 +121,15 @@ void random_standard_exponential_inv_fill(bitgen_t * bitgen_state, npy_intp cnt,
 {
   npy_intp i;
   for (i = 0; i < cnt; i++) {
-    out[i] = -log(1.0 - next_double(bitgen_state));
+    out[i] = -npy_log1p(-next_double(bitgen_state));
```

Comment:
I think never change anything since the next_double is always at least `1.1102230246251565e-16` when non-zero, and so these two are identical.  In this sense all of these are "free" to change without worrying about streams.